### PR TITLE
Use client.rack property in kafka connect

### PR DESCRIPTION
**Pull request by klalafaryan**
Signed-off-by: klalafaryan <lalafaryan@gmail.com>

### Type of change

- Enhancement / new feature

### Description

This PR adds new property `client.rack` for kafka connect cluster, which allows to use `fetch from closest replica` feature. 

It is covering #3426 ticket.

### Checklist

- [x] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [x] Reference relevant issue(s) and close them after merging
- [x] Update CHANGELOG.md

### CODE REVIEW

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaConnectClusterTest.java [hunk 1]
```diff
@@ -1346,4 +1337,93 @@ public void testNetworkPolicyWithoutConnectorOperator() {
...
+        checkOwnerReference(kc.createOwnerReference(), dep);
+        checkRack(dep, resource);
+    }
+
+    private void checkOwnerReference(OwnerReference ownerRef, HasMetadata resource)  {
+        assertThat(resource.getMetadata().getOwnerReferences().size(), is(1));
+        assertThat(resource.getMetadata().getOwnerReferences().get(0), is(ownerRef));
+    }
+
+    private void checkRack(Deployment deployment, KafkaConnect resource) {
```
[the remark this answers is not in the document]
Comment (klalafaryan) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaConnectClusterTest.java:1412]: I have already the test method with `testGenerateDeploymentWithRack` name. This is only assertion of the rack specific properties. I couldn't find the better name, and just to be consistent with other methods I use `checkRack`. Any suggestions ?

[the remark this answers is not in the document]
Comment (scholzj) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaConnectClusterTest.java:1412]: Never mind this comment. I thought it is never called form `testGenerateDeploymentWithRack`. But now I see it is called from `checkDeployment` which is called there. So just ignore this. Sorry.

**Review by scholzj — COMMENTED** [review]
LGTM. Thanks for the PR.

**Comment by see-quick** [comment]
Would be really great to write some system test for it. This is just referenced from our chat on Slack. 
Test case that i imagine:
1. Deploy Strimzi Operator
2.Deploy Kafka cluster (any preferences how many replicas the Kafka should have?)
3.Deploy Kafka connect (with client.rack  property)
4.Send some messages to the Kafka cluster
5.Create for instance FileSink connector (to consume data from. specific topic)
6.Find out what is the closest replica?
7.Verify that Kafka Connect reads data from the closest replica.

**Review by ppatierno — APPROVED** [review]
Thanks for the PR! LGTM. Just fix the conflicts please.

**Comment by scholzj** [comment]
`testRecoveryFromKafkaStatefulSetDeletion` test is flaky - there is another PR to fix that. So that has nothing to do with this PR.

@klalafaryan @see-quick: I decided to merge this PR since you are off and we would just get more conflicts. It would be great if you can both get back to the system tests once you get back.

**Review by scholzj — APPROVED** [review]

**Comment by scholzj** [comment]
BTW: Thanks for the PR @klalafaryan

**Pull request merged**